### PR TITLE
Change headerName type to utf-8 text

### PR DESCRIPTION
**Pull request by epicallan**

### CODE REVIEW

File: xray/test/stack-9.0-dependencies.txt [hunk 1]
```diff
@@ -126,7 +126,7 @@ integer-gmp 1.1
 integer-logarithms 1.0.3.1
 invariant 0.5.4
 kan-extensions 5.2.3
-lambda-alb 0.0.1
+lambda-alb 0.0.2
```
Comment (epicallan) [inline xray/test/stack-9.0-dependencies.txt:129]: I can't seem to get why this `lambda-alb`  as part of `x-ray` dependencies list.

Comment (mbj) [reply xray/test/stack-9.0-dependencies.txt:129]: bug in xray devtool setup, does not constrain, misses explicit target.

File: lambda-alb/src/AWS/Lambda/Header.hs [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
 import qualified Data.Aeson.Types     as JSON
 import qualified Data.CaseInsensitive as CI
 import qualified Data.HashMap.Strict  as HashMap
 import qualified Data.Text.Encoding   as Text
-import qualified Network.HTTP.Types   as HTTP
 
-newtype Headers = Headers [HTTP.Header]
+type HeaderName = CI.CI Text
+
+type Header = (HeaderName, ByteString)
```
Comment (mbj) [inline lambda-alb/src/AWS/Lambda/Header.hs:15]: I think we should not stop here and alos make the value a `Text` as ALB does not return byte strings.

File: lambda-alb/src/AWS/Lambda/Header.hs [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
+
+newtype Headers = Headers [Header]
   deriving stock (Eq, Show)
 
 instance JSON.ToJSON Headers where
   toJSON (Headers list)
     = JSON.Object . HashMap.fromList
-    $ bimap (decodeUtf8 . CI.original) (JSON.String . decodeUtf8) <$> list
+    $ bimap CI.original (JSON.String . decodeUtf8) <$> list
```
Comment (mbj) [inline lambda-alb/src/AWS/Lambda/Header.hs:23]: @epicallan With: https://github.com/mbj/mhs/pull/94/commits/67982a5ce56075dd1bf5ad7ea9e4172e4b15d55b#r792298712 we'd get a rid of the other `decodeUtf8` which is a partial function.

File: lambda-alb/src/AWS/Lambda/Header.hs [hunk 4]
```diff
@@ -35,3 +35,10 @@ instance JSON.FromJSON Headers where
           (pure . Text.encodeUtf8)
           value
         pure (CI.mk headerName, headerValue)
+
+hAccept,hAuthorization, hCacheControl, hContentType, hOrigin :: HeaderName
```
Comment (dkubb) [inline lambda-alb/src/AWS/Lambda/Header.hs:39]: __suggestion: (nitpick)__ There's a missing space after the `,`.

**Pull request closed without merging**